Support version tags that don't start with v for semantic versioning

davidmfinol opened the pull request:
#### Changes

Fixes #242

#### Checklist

<!-- please check all items and add your own -->

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

GabLeRoux commented:
I totally agree with this. According to https://semver.org/ prefixing with `v` isn't considered semantic so allowing versions without it should definitely be considered 😅 

> Is “v1.2.3” a semantic version?
>
> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

webbertakken commented:
> I totally agree with this. According to https://semver.org/ prefixing with `v` isn't considered semantic so allowing versions without it should definitely be considered 😅
> 
> > Is “v1.2.3” a semantic version?
> > No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control. Example: git tag v1.2.3 -m "Release version 1.2.3", in which case “v1.2.3” is a tag name and the semantic version is “1.2.3”.

While I agree we should support this, the `v` is not specific to `semantic versioning` and not part of the semantic version itself. It is GitHub convention to use `v` in the tag to indicate it's a version tag.

webbertakken commented on src/model/versioning.test.ts:
```diff
@@ -125,6 +125,7 @@ describe('Versioning', () => {
       'is happy with valid %s',
       (description) => {
         expect(Versioning.descriptionRegex1.test(description)).toBeTruthy();
+        expect(Versioning.descriptionRegex1.test(description?.slice(1))).toBeTruthy();
```
Suggest adding a separate tests for `it('accepts valid semantic versions without v-prefix', () => {})`

davidmfinol replied:
Done

webbertakken commented:
Two more things:

- Would you mind updating semver-regex as part of this PR?
![image](https://user-images.githubusercontent.com/20756439/145723819-3c9850a5-e6ef-4532-9b5c-8f311fac27b2.png)
- Could you please use descriptive PR titles, so that the releases properly show what changed (without manual work). See description generator: https://takken.io/tools

davidmfinol commented:
> * Would you mind updating semver-regex as part of this PR?
I didn't want to manually update the yarn.lock file, so I found that the semver-regex dependency came from husky, and I updated the versions of husky and lint-staged. The semver-regex dependency is now gone.

> * Could you please use descriptive PR titles, so that the releases properly show what changed (without manual work). See description generator: https://takken.io/tools
I was hoping that the name would come from the linked issue, but I guess it doesn't do that. I've updated the title.

webbertakken commented on .husky/.gitignore:
```diff
@@ -0,0 +1 @@
+_
```
what does this do exactly?

davidmfinol replied:
Yeah, it's auto-generated. The `husky/_` folder seems to be generated during the pre-commit check as a temporary workspace
